### DISCUSSION TITLE
J2534 (passthrucan) connection does not stay connected #208

### DISCUSSION THREAD

**Discussion by Tmod700** [collin80/SavvyCAN#560] [Q&A]
I got x64 Build 208 to show the port selections when choosing a passthru connection but it connects and then disconnects itself just a few seconds later. I am using Tactrix Openport 2.0 and Win10x64 and downloaded the 208 CIBuild.
I tried 207 compiled for 32bit and it connects and stays connected fine.
Any ideas as to why it auto disconnects?
![Ashampoo_Snap_2023 01 01_12h08m01s_001_](https://user-images.githubusercontent.com/79778536/210200526-3de0ad7d-fe27-478d-90aa-5ace5d59a737.png)

**Comment by collin80** [comment]
J2534 support has never worked right. I think I bodged it into working once but have yet to release a version that actually works. I've created a GitHub issue out of this because it certainly needs to be fixed.

**Reply by Tmod700** [reply]
collin, I certainly want to thank you for taking the time to keep this project alive and after many trial runs with different software out there this has the most promise.
I wish I knew coding as I would most certainly help but it is way over my head.
Hopefully there is a easy solution to this issue.
Thanks again!

**Comment by Bill400** [comment]
I've fresh loaded WIN7_64 (Panasonic OEM) on a CF-53 mk2, then SavvyCAN V208 and Kvaser driver (they're all the same) for a 2XCAN/LIN hybrid. Initally, no communications, though SavvyCAN connection window reported "connected".
After much gnashing of teeth, went to Device Manager and disabled "Virtual Device" and acheived communications.
New problem: connection defaults to 250kbs, even after change/save/reset in connection window.
As soon as any SavvyCAN function is invoked, driver defaults to 250kbs and comm halts.
A post on the old website said Collin hacked the device driver to change the default.
If V207_32 works, I'll give that a try.
Collin: Thank you for your work and for sharing SavvyCAN. 
Like others, my objective is to RE a vehicle as part of a larger project (AT to MT conversion).

**Reply by Bill400** [reply]
Upgraded again to WIN10_64 on the Panasonic CF-53 Mk2. Using Visual Studio 2022 to examine the V208 source files. 
Connections module\Canbus.cpp appears to contain our default bus speed of 250Kbts. Statements that follow appear to set the operating parameters. Not yet the C++ expert, I wonder about the last two statements, which are identical except for the const keyword which specifies that the object or variable isn't modifiable. Is the final statement without the const keyword an error? Should it have been deleted?
PS: I didn't notice the bit shift function difference, one left, one right.

Since I experienced a recurring default to 250Kbts , a work around (for 500Kbts) might be to change the default 250000 to 500000?

**Comment by ptalev** [comment]
Hi i want use with j2534 sm2 usb on win 10x64 but at new connection set no device in drop down menu port and not connected
Thanks advance

**Reply by eyal32593-star** [reply]
hi dear
i have same issu with sm3
have you solve it?

**Comment by dx168b** [comment]
Passthru not working if SavvyCAN builded as x64 version. All J2534 drivers for all VCI is x86 version.
Also SavvyCAN not capturing extended frames format (29 bit CAN ID) via Passthru. This bug need to fix.
